### ISSUE TITLE
OC Screens do not refresh after Server Restart

### ISSUE THREAD

**Issue by Merung-Kelsith** [MightyPirates/OpenComputers#2962]
MC Version:  1.12.2
OC Version:  OpenComputers-MC1.12.2-1.7.2.67
Screen:  6x Tier 3 screens, 2 high, 3 across
GPU:  Tier 3

TL;DR version:  Is there a way to refresh an OC screen manually?

Long version:
I have multiple computers running to do things like control my DE reactor, monitor my DE storage, autocraft, etc.  Everything works just fine when I start the programs.  Since I run a private server on AWS, I shut it down whenever we are not logged in to save on cost.

When I restart the server, though, the OC screens do not refresh.  They stay static, and do not update.  The programs are still running on the back end.  I can tell this because the program that controls the reactor has buttons to change settings, and pressing those buttons (even with the screen frozen) changes the values.  I experiment by changing the field strength and verifying by looking at the reactor that the values have changed (they have).

The only way I can find to get the screens to start refreshing again is to completely shut down the computer and restart it.  While this is fine for most of the computers, restarting the reactor control comp  makes the reactor explode if I do not shut the reactor down first... but if I do that it takes hours for the saturation to drain to a point I can start it again.

So, Is there a way to force a refresh/redraw of the OC screen without completely shutting down the computer?

**Comment by payonel** [comment]
First of all, this shouldn't happen. The screens should stay connected and active between server restarts.

Tell me about these buttons. Are they rendered buttons you've programmed on the screen? Are you able to "open" the screen (a screen with a keyboard can "open"), and send "touch" or "key_*" events to the machine?

Have you tried removing the screens and placing them again? (this shouldn't be necessary, but might mitigate the issue)

what version of mc and what version of oc is this?

are the screens located in separate chunks than the machine case?

are you using enderio conduits in blocks adjacent to the machines, cables, or screens?

**Comment by Merung-Kelsith** [comment]
The program I use with the buttons is this one:  https://pastebin.com/t9ETGyZk 

I am very new to OC, so I have not gotten into my own button programming yet.  I just used that to test if the program was still running on the back end.  This happens with all 3 of my computers though, with or without buttons.

Yes, I have tried removing/replacing the screens.  When I removed one (or more) that were not directly connected to the computer, the frozen screen reappeared on the 2x2 that was left.  When I tried removing the screen that was connected to the computer, the program stopped and my reactor exploded (luckily I backed it up before doing anything).

MC Version: 1.12.2
OC Version: OpenComputers-MC1.12.2-1.7.2.67

No, I have the 2 high by 3 wide screen, and the computer is on top of the middle screen

Yes, I am using EnderIO OC Conduits to connect to the OC Adapters that connect to the other components (On the computer side, the conduits attach to one of the screens, not the computer itself.  Could this be an issue?)

Comp 1:  Draconic Reactor, 2x flux gates
Comp 2:  Draconic Energy Storage
Comp 3: ME System

The 3 comps are NOT connected to each other

Thanks for the quick response!

**Comment by payonel** [comment]
I understand now that the buttons are rendered from software in the OC machine, thanks for the program link

how are you testing the buttons work though? Do you punch the screen? Do you "open" the screen and then "touch" the button?

Yes, the buttons are not at all part of the root cause, i just wanted to know what these buttons were. we're good there though, now I understand

the screen removal process isn't a great mitigation, there are going to be side effects. I could explain other methods of tinkering with screens, but it isn't a solution to the problem, so it might just add complexity to this discussion that isn't helpful

we had issues with enderio conduits in the past, but that should be fixed in 1.7.2 (which you are using)

If you are able to, please reach out to me in irc (espernet #oc ) or discord ( https://discord.gg/UTcZvj )

**Comment by Merung-Kelsith** [comment]
For the buttons, I open the screen and then touch the button.  For the rest, I have joined the Discord server to continue the discussion there.  Thanks!

**Comment by payonel** [comment]
closed by 1e35f8e49ac7500fcac1380d5ba38b7b93f5e7bf

**Comment by payonel** [comment]
closed by 1e35f8e49ac7500fcac1380d5ba38b7b93f5e7bf

**Comment by payonel** [comment]
github is broken...this is the fourth time i've tried to close this ticket
this is the commit that closes this ticket: 1e35f8e49ac7500fcac1380d5ba38b7b93f5e7bf
